PyTorch/diffusion/sd/app.py

```diff
@@ -0,0 +1,99 @@
+import torch
+import torch_directml
+import gradio as gr
+from diffusers import AutoPipelineForText2Image,  StableDiffusionPipeline, LMSDiscreteScheduler
+from PIL import Image
+import numpy as np
+ 
+def preprocess(image):
+    w, h = image.size
+    w, h = map(lambda x: x - x % 32, (w, h))  # resize to integer multiple of 32
+    image = image.resize((w, h), resample=PIL.Image.LANCZOS)
+    image = np.array(image).astype(np.float32) / 255.0
+    image = image[None].transpose(0, 3, 1, 2)
+    image = torch.from_numpy(image)
+    return 2.*image - 1.
```

Comment:
```suggestion
    return 2. * image - 1.
```
spacing saves ~~lives~~ eyes.